Objects/genobject.c

```diff
@@ -397,9 +396,10 @@ gen_close(PyGenObject *gen, PyObject *args)
     _PyInterpreterFrame *frame = (_PyInterpreterFrame *)gen->gi_iframe;
     /* It is possible for the previous instruction to not be a
      * YIELD_VALUE if the debugger has changed the lineno. */
-    if (err == 0 && is_yield(frame->prev_instr)) {
-        assert(is_resume(frame->prev_instr + 1));
-        int exception_handler_depth = frame->prev_instr[0].op.code;
+    if (err == 0 && is_yield(frame->instr_ptr - 1)) {
```

Comment:
Ugh. If I believe the comment, this still has the problem (not new) that it peeks back at whatever came before, which might be a cache entry with some arbitrary value.

Reply:
And if I try to fix the optimization (https://github.com/python/cpython/pull/111069) then some tests fail.

Reply:
From the comment a few lines back it looks like the previous instruction normally *is* YIELD_VALUE, so the only way it can become an issue is if the user uses the debugger's goto/jump feature *and* jumps to something preceded by an unfortunate cache entry. Let's just add something to that comment to the extent that this isn't 100% safe *if* a jump/goto was used.

Reply:
I created https://github.com/python/cpython/issues/111107 to deal with this properly.